src/dialect/mssql.rs

```diff
@@ -215,6 +225,59 @@ impl MsSqlDialect {
         }))
     }
 
+    /// Parse `CREATE TRIGGER` for [MsSql]
+    ///
+    /// [MsSql]: https://learn.microsoft.com/en-us/sql/t-sql/statements/create-trigger-transact-sql
+    fn parse_create_trigger(
+        &self,
+        parser: &mut Parser,
+        or_alter: bool,
+    ) -> Result<Statement, ParserError> {
+        let name = parser.parse_object_name(false)?;
+        parser.expect_keyword_is(Keyword::ON)?;
+        let table_name = parser.parse_object_name(false)?;
+        let period = parser.parse_trigger_period()?;
+        let events = parser.parse_comma_separated(Parser::parse_trigger_event)?;
+
+        parser.expect_keyword_is(Keyword::AS)?;
+
+        let trigger_statements_body = if parser.peek_keyword(Keyword::BEGIN) {
+            let begin_token = parser.expect_keyword(Keyword::BEGIN)?;
+            let statements = parser.parse_statement_list(&[Keyword::END])?;
+            let end_token = parser.expect_keyword(Keyword::END)?;
+
+            BeginEndStatements {
+                begin_token: AttachedToken(begin_token),
+                statements,
+                end_token: AttachedToken(end_token),
+            }
+        } else {
+            BeginEndStatements {
+                begin_token: AttachedToken::empty(),
+                statements: parser.parse_statements()?,
+                end_token: AttachedToken::empty(),
+            }
```

Comment:
Oh I think instead of representing the begin optionality using tokens, we can use an enum, something similar to [ConditionalStatements](https://github.com/apache/datafusion-sqlparser-rs/blob/main/src/ast/mod.rs#L2294-L2299) where we have an explicit `BeginEnd` enum variant

Reply:
Gotcha, yeah I could see it either way. I will update this to use the enum pattern

Reply:
Done 👍